operator/pkg/ciliumidentity/identity.go

```diff
@@ -3,8 +3,109 @@
 
 package ciliumidentity
 
-import "github.com/cilium/cilium/pkg/k8s/resource"
+import (
+	"context"
+	"time"
+
+	"k8s.io/client-go/util/workqueue"
+
+	ciliumapiv2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
 
 func cidResourceKey(cidName string) resource.Key {
 	return resource.Key{Name: cidName}
 }
+
+func (c *Controller) processCiliumIdentityEvents(ctx context.Context) error {
+	for event := range c.ciliumIdentity.Events(ctx) {
+		switch event.Kind {
+		case resource.Upsert:
+			c.logger.Debug("Got Upsert CID event",
+				logfields.CIDName, event.Key.String())
+			c.onCiliumIdentityUpsertEvent(event.Object)
+		case resource.Delete:
+			c.logger.Debug("Got Delete CID event",
+				logfields.CIDName, event.Key.String())
+			c.onCiliumIdentityDeleteEvent(event.Object)
```

Comment:
We trigger reconciliation with no distinction for either upsert/delete thus why do we need two separate methods? vs something like reconcile(ResourceKey, delay). 
It might be worth evaluating the design choices here so we can dry up this code and streamline the needs to reimplement similar logic for namespace, pod and cids, ... 
Perhaps leveraging generics or tidying the interfaces could help us dry this up a little more.

Reply:
Indeed, we do not need two separate methods, moved both under the same method. We still need to have different queues for CIDs and Pods as we need to take different actions. For pods we end up with `reconcilePod` while for CIDs we end up with `reconcileCID`.  

We could potentially refactor the existing implementation of the queues and have a single queue and rework the existing worker logic to differentiate based on the resource key, but I'm not sure how much value it would bring.

Reply:
Indeed, I gave I try to your suggestion and code looks much better, I extracted it to a separate interface. I think we might just need to spawn two workers now to archive the same effect. But for now let's go with only one and we can easily adjust later to have a configurable number of workers.